src/workers/notifications/utils.ts

```diff
@@ -41,18 +41,50 @@ export async function articleNewCommentHandler(
   con: DataSource,
   commentId: string,
 ): Promise<NotificationHandlerReturn> {
-  const comment = await con
-    .getRepository(Comment)
-    .findOne({ where: { id: commentId }, relations: ['user'] });
+  const repo = con.getRepository(Comment);
+  const comment = await repo.findOne({
+    where: { id: commentId },
+    relations: ['user'],
+  });
   if (!comment) {
     return;
   }
   const postCtx = await buildPostContext(con, comment.postId);
   if (!postCtx) {
     return;
   }
+  const excludedUsers = [comment.userId];
+
+  const isReply = !!comment.parentId;
+  if (isReply && (postCtx.post.authorId || postCtx.post.scoutId)) {
+    const threadFollower = await repo
+      .createQueryBuilder()
+      .select('"userId"')
+      .where({
+        id: comment.parentId,
+        userId: postCtx.post.authorId,
+      })
+      .orWhere({
+        parentId: comment.parentId,
+        userId: postCtx.post.authorId,
+      })
+      .orWhere({
+        id: comment.parentId,
+        userId: postCtx.post.scoutId,
+      })
+      .orWhere({
+        parentId: comment.parentId,
+        userId: postCtx.post.scoutId,
+      })
```

Comment:
I suggest a simpler condition. `(id == comment.parentId || parentId == comment.parentId) && (userId == authorId || userId == scoutId)`. Also make sure to group by `userId` to make sure you don't get the same user multiple times

Reply:
I have now updated the conditions to be much simpler. Would love to hear what you think 🚢